### DISCUSSION TITLE
Scores with "Fun" mods enabled still get submitted to leaderboard

### DISCUSSION THREAD

**Discussion by WiIIiamWei** [ppy/osu#27719] [Ideas]
### Type

Other

### Bug description

I was checking the leaderboard of [TheFatRat - Mayday (feat. Laura Brehm)](https://osu.ppy.sh/beatmapsets/756794#osu/1605148) and saw this No.1 score:
![image](https://github.com/ppy/osu/assets/37211959/0e4e5b97-b427-45c5-9a69-915c8310ea2f)
Link: https://osu.ppy.sh/scores/2487185020
I don't think this should be appeared in the leaderboard, as a lot of UNRANKED mods are selected. I initially thought this is an intentional behavior as it's a LOVED map, so I tried it out on a recently RANKED map with Muted enabled, and my score appeared in the leaderboard (although unranked).
<img width="766" alt="image" src="https://github.com/ppy/osu/assets/37211959/c9240827-49e6-46d7-87c4-c5ff5e588784">
<img width="684" alt="image" src="https://github.com/ppy/osu/assets/37211959/60d6c26a-01d6-44c4-951c-92c01fead1c6">
*Note: There are delays between two screenshots. That's the reason why they're in different ranks.*
My score: https://osu.ppy.sh/scores/2553132296 (I suck at clicking circles)
I assume this could mess up the leaderboard in Lazer mode and cause confusions. **Submitting** UNRANKED play is completely fine, but I don't think is the same case for **leaderboards**. Please tell me whether this is intentional or not.

### Screenshots or videos

As shown above.

### Version

2024.312.1

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14731313/compressed-logs.zip)

**Comment by Gabixel** [comment]
Unranked mods just prevent you from getting pp, but they can appear in the leaderboard. I don't know what exactly the ultimate goal is yet on how to show a lot of mods in the leaderboards, but I believe it's all intentional

**Comment by WiIIiamWei** [comment]
I don't think it's a good idea though. Using cursed mod combinations could dramatically decrease the difficulty, just like the No.1 score on Mayday (you can watch the replay, it literally didn't need to aim). Players could potentially abuse this feature to kill the leaderboard.

**Comment by Stoppedpuma** [comment]
> Using cursed mod combinations could dramatically decrease the difficulty

Magnetised is the only mod which make scores like the one on mayday a thing, this shouldn't matter for leaderboards once its multiplier starts to change based on the amount of attraction.

Removing an entire category of mods over an issue which will be fixed is stupid. Players abusing these mods to "kill the leaderboard" isn't a concern since multipliers are still a work in progress and can always be adjusted. These mods are unranked for a reason.

**Comment by Liam-DeVoe** [comment]
It's intentional that unranked mods show on the leaderboard. Arguably the magnetised mod multiplier should be reduced though.

**Comment by bdach** [comment]
https://github.com/ppy/osu/pull/27724

**Comment by WiIIiamWei** [comment]
Haven't seen this discussion for a while and it has updated a lot, I appriciate everyone's answer.

However the discussion in #27724 did not come with an agreement, so I think it's better to leave this thread open (and if necessary, discuss here instead of in the PR), also to avoid the situation where another player is confused like me and mentions this in a new issue or discussion.

As a player who does not play very much and very well, I think [Kori3's comment](https://github.com/ppy/osu/pull/27724#issuecomment-2022321678) has a point, in which "Ranked" and "Unranked", even though not the actual case, really sound more intuitive. A better name should it be if the change is necessary.

Maybe consider the following solution (also a complement to [peppy's](https://github.com/ppy/osu/pull/27724#issuecomment-2017741493)), in order to meet different needs:

- Add multipliers to some fun mods (I suppose it needs to be done anyway).
- Add a checkbox which says something like "Display Fun Mods" and filter the leaderboard accordingly. Or just disabling them all if no one likes it.

~~Also worth mentioning, should PP be calculated/displayed when "Fun" mods are enabled? Since they are "not eligible for PP", why not just don't let them appear in the first place?~~
![image](https://github.com/ppy/osu/assets/37211959/832be860-f039-4676-bf0f-bb08c5a15f6e)

**Reply by bdach** [reply]
> Also worth mentioning, should PP be calculated/displayed when "Fun" mods are enabled? Since they are "not eligible for PP", why not just don't let them appear in the first place?

Muted is a ranked / "eligible for PP" mod.

**Reply by WiIIiamWei** [reply]
Sorry I didn't know. Thanks for correction.

**Comment by WiIIiamWei** [comment]
> Add a checkbox which says something like "Display Fun Mods" and filter the leaderboard accordingly. Or just disabling them all if no one likes it.

> Muted is a ranked / "eligible for PP" mod.

In that case this needs reconsideration, what should be displayed and what should not. I originally thought all "Fun" mods are "not eligible for PP". Sorry for the confusion.

**Reply by bdach** [reply]
The set of "ranked" mods is not final. Right now we're limited in what we _can_ enable, but that does not mean that no more mods will be allowed in some indeterminate future.

**Reply by WiIIiamWei** [reply]
Should I say that "Fun" mods are more like "Experimental" mods?
Cuz "Fun" sounds like it's "just for fun" only and "not eligible for PP"... dunno how should I put this...
It's just my own thoughts though.

**Reply by Rekunan** [reply]
I can agree that the category name isn't rly descriptive, and I think experimental would have its own set of connotations, tho I'm having trouble coming up with anything else. Maybe just take from approach different and change how that mod works (or named).
